crypto/des/fcrypt.c

```diff
@@ -70,9 +70,12 @@ char *DES_crypt(const char *buf, const char *salt)
     char e_buf[32 + 1];         /* replace 32 by 8 ? */
     char *ret;
 
+    if (sizeof(salt) < 2) return NULL;
+    if (salt[0] == '\0' || salt[1] == '\0') return NULL;
```

Comment:
It was "valid", with one size length array, but further else the salt[1] value was not set, so its content was not defined. And Linux crypt() function, requires that salt value is at least 2 char size array.